File: project/5.03_コンストラクタの追加/java/java_ex5_03_2/src/java_ex5_03_2/Ex5_03_2.java
package java_ex5_03_2;

class Sphere {
	double radius;					// 半径
//	double[] cp = new double[2];	// 中心座標
	double[] cp;					// 中心座標
	
	Sphere( double arg_radius, double[] arg_cp ){
		radius = arg_radius;
		cp = arg_cp;
	}
}

public class Ex5_03_2 {

	public static void main(String[] args) {
		
//		Sphere cube = new Sphere();
//		cube.radius = 5.5;
//		cube.cp = new double[] {1.2,3.4};
		
		Sphere cube = new Sphere( 5.5, new double[] {1.2, 3.4} );

		System.out.println( "半径：" + cube.radius );
		System.out.println( "中心座標：" + cube.cp[0] + "," +  cube.cp[1] );
		
	}

}

File: project/5.03_コンストラクタの追加/py/py_ex5_03_2/src/ex5_03_2.py
'''
Created on 2023/04/30

@author: Keroichi.T
'''

class Sphere:
    
    def __init__(self, arg_radius, arg_cp ):
        self.radius = arg_radius
        self.cp = arg_cp
 
def main():
    
    cube = Sphere( 5.5, [1.2, 3.4] )
    
    print( "半径：{}".format( cube.radius ) )
    print( "中心座標：{},{}".format( cube.cp[0], cube.cp[1] ) )
    
if __name__ == '__main__':
    main()